funcx_endpoint/funcx_endpoint/endpoint/register_endpoint.py

```diff
@@ -0,0 +1,58 @@
+import os
+import json
+
+import funcx_endpoint
+
+
+def register_endpoint(logger, funcx_client, endpoint_uuid, endpoint_dir, name):
+    """Register the endpoint and return the registration info. This function needs
+    to be isolated (including the logger which is passed in) so that the function
+    can both be called from the endpoint start process as well as the daemon process
+    that it spawns.
+
+    Parameters
+    ----------
+    logger : Logger
+        Logger to use
+
+    funcx_client : FuncXClient
+        The auth'd client to communicate with the funcX service
+
+    endpoint_uuid : str
+        The uuid to register the endpoint with
+
+    endpoint_dir : str
+        The endpoint directory path to store data in
+
+    name : str
+        The name of the endpoint
+    """
+    logger.debug("Attempting registration")
+    logger.debug(f"Trying with eid : {endpoint_uuid}")
+    reg_info = funcx_client.register_endpoint(name,
+                                              endpoint_uuid,
+                                              endpoint_version=funcx_endpoint.__version__)
+
+    # this is a backup error handler in case an endpoint ID is not sent back
+    # from the service or a bad ID is sent back
```

Comment:
We should add a note that only the endpoint id is reused from the endpoint.json file. New forwarder URI are fetched everytime we start and register.